test/sol6/reserves/kyberFprReserveV2.js

```diff
@@ -0,0 +1,2902 @@
+const Reserve = artifacts.require("KyberFprReserveV2");
+const MockSanityRates = artifacts.require("MockSanityRates");
+const NoPayableFallback = artifacts.require("NoPayableFallback");
+
+const Helper = require("../../helper.js");
+const BN = web3.utils.BN;
+
+//global variables
+//////////////////
+const {BPS, precisionUnits, ethDecimals, ethAddress, zeroAddress, 
+    zeroBN, MAX_QTY, MAX_RATE, MAX_ALLOWANCE} = require("../../helper.js");
+const {expectEvent, expectRevert} = require('@openzeppelin/test-helpers');
+
+//balances
+let expectedReserveBalanceWei = new BN(0);
+let expectedReserveBalanceWeth = new BN(0);
+let reserveTokenBalance = [];
+let reserveTokenImbalance = [];
+
+//permission groups
+let admin;
+let operator;
+let alerter;
+let network;
+let withdrawAddress;
+
+//contracts
+let weth;
+let convRatesInst;
+let reserveInst;
+let sanityRate;
+
+//block data
+let currentBlock;
+let validRateDurationInBlocks = 1000;
+let maxGasPrice = new BN(150).mul(new BN(10).pow(new BN(9))); // 150 * 10^9
+
+//tokens data
+////////////
+let numTokens = 5;
+let tokens = [];
+let tokenDecimals = [];
+let tokenAdd = [];
+
+// imbalance data
+let minimalRecordResolution = new BN(2);
+let maxPerBlockImbalance = 40000;
+let maxTotalImbalance = maxPerBlockImbalance * 12;
+
+// all price steps in bps (basic price steps).
+// 100 bps means rate change will be: price * (100 + 10000) / 10000 == raise rate in 1%
+// higher rate is better for user. will get more dst quantity for his tokens.
+// all x values represent token imbalance. y values represent equivalent steps in bps.
+// buyImbalance represents coin shortage. higher buy imbalance = more tokens were bought.
+// generally. speaking, if imbalance is higher we want to have:
+//      - smaller buy bps (negative) to lower rate when buying token with ether.
+//      - bigger sell bps to have higher rate when buying ether with token.
+////////////////////
+
+//base buy and sell rates (prices)
+let baseBuyRate = [];
+let baseSellRate = [];
+
+
+//imbalance buy steps
+let imbalanceBuyStepX = [-85000, -28000, -15000, 0, 15000, 28000, 45000];
+let imbalanceBuyStepY = [ 1300, 130, 43, 0, 0, -110, -160, -1600];
+
+
+//sell imbalance step
+let imbalanceSellStepX = [-85000, -28000, -10000, 0, 10000, 28000, 45000];
+let imbalanceSellStepY = [-1500, -320, -75, 0, 0, 110, 350, 650];
+
+
+//compact data.
+let sells = [];
+let buys = [];
+let indices = [];
+let compactBuyArr = [];
+let compactSellArr = [];
+
+contract('KyberFprReserveV2', function(accounts) {
+    before("Global setup", async function () {
+        // set account addresses
+        admin = accounts[0];
+        operator = accounts[1];
+        network = accounts[2];
+        user1 = accounts[4];
+        user2 = accounts[5];
+        withdrawAddress = accounts[6];
+        sanityRate = accounts[7];
+        alerter = accounts[8];
+        walletForToken = accounts[9];
+        user3 = accounts[8];
+        user4 = accounts[9];
```

Comment:
I think we use 20 accounts by defaults
in the config file.

so can use them if needed. thus don't have to reuse accounts.